wslib.h

```diff
@@ -0,0 +1,100 @@
+
+/* Example of use
+
+
+
+                  void main(){
+
+                              char tmp_name[200];
+                              strcpy(tmp_name,"styl.css");
+
+                              struct safeObject myObject;
+
+                              createSafeObjectFromFile(&myObject,tmp_name);
+
+                              char *ptr=myObject.sObjectStart;
+                              for(int i=0;i<myObject.sSize;i++){
+                              printf("%c",*ptr);ptr++;}
+
+
+                              ptr=myObject.sObjectStart;
+                              free(ptr);} */
+
+char safeChars[] = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k',
+                    'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
+                    'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D', 'E', 'F', 'G',
+                    'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R',
+                    'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', '0', '1', '2',
+                    '3', '4', '5', '6', '7', '8', '9', '-', '_'};
+
+long long getFileSize(const char *fileName) {
+  FILE *file = fopen(fileName, "rb");
+  if (!file) {
+    perror("Could not open the file");
+    return -1; // Wartość -1 oznacza błąd
+  }
+
+  fseek(file, 0, SEEK_END);
+  long long fileSize = ftell(file);
+  fclose(file);
+
+  return fileSize;
+}
+
+char getRandomSafeChar() {
+  const char charset[] =
+      "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  return charset[rand() % (sizeof(charset) - 1)];
+}
+
+char getRandomChar() { return rand() % 126; }
+
+struct safeObject {
+  long long sSize;
+  char *sObjectStart;
+  char *sObjectEnd;
+};
+
+int getSafeCharCode(char *ptr) {
+  char nr_1, nr_2;
+  for (int i = 0; i < 70; i++) {
+    if (*ptr == safeChars[i]) {
+      nr_2 = i;
+      break;
+    }
+  }
+  ptr++;
+  for (int i = 0; i < 70; i++) {
+    if (*ptr == safeChars[i]) {
+      nr_1 = i;
+      break;
+    }
+  }
+  return (nr_1 << 4) | (nr_2 & 0x0F);

```

Comment:
Czy to robi jakieś pole bitowe po 16 bitów na każde z `nr_1` i `nr_2`? Przydałby się jakiś komentarz, który to opisuje.